docs/tutorials/azureml.md

```diff
@@ -0,0 +1,335 @@
+    else:
+        results['error'] = "I am unable to find the answer to this question. Can you please ask another question?"
+    return results
+
+# Perform the one-off intialisation for the prediction. The init code is run once when the endpoint is setup.
+def init():
+    global tokenizer, session, model
+
+    model_name = "bert-large-uncased-whole-word-masking-finetuned-squad"
+    model = transformers.BertForQuestionAnswering.from_pretrained(model_name)
+
+    # use AZUREML_MODEL_DIR to get your deployed model(s). If multiple models are deployed, 
+    # model_path = os.path.join(os.getenv('AZUREML_MODEL_DIR'), '$MODEL_NAME/$VERSION/$MODEL_FILE_NAME')
+    model_dir = os.getenv('AZUREML_MODEL_DIR')
+    if model_dir == None:
+        model_dir = "./"
+    model_path = os.path.join(model_dir, model_name + ".onnx")
+
+    # Create the tokenizer
+    tokenizer = transformers.BertTokenizer.from_pretrained(model_name)
+
+    # Create an ONNX Runtime session to run the ONNX model
+    session = onnxruntime.InferenceSession(model_path, providers=["CPUExecutionProvider"])  
+
+
+# Run the PyTorch model, for functional and performance comparison
+def run_pytorch(raw_data):
+    inputs = json.loads(raw_data)
+
+    model.eval()
+
+    logging.info("Question:", inputs["question"])
+    logging.info("Context: ", inputs["context"])
+
+    input_ids, input_mask, segment_ids, tokens = preprocess(inputs["question"], inputs["context"])
+    model_outputs = model(torch.tensor([input_ids]),  token_type_ids=torch.tensor([segment_ids]))
+    return postprocess(tokens, model_outputs.start_logits.detach().numpy(), model_outputs.end_logits.detach().numpy())
+
+# Run the ONNX model with ONNX Runtime
+def run(raw_data):
+    logging.info("Request received")
+    inputs = json.loads(raw_data)
+    logging.info(inputs)
+
+    # Preprocess the question and context into tokenized ids
+    input_ids, input_mask, segment_ids, tokens = preprocess(inputs["question"], inputs["context"])
+    logging.info("Running inference")
+    
+    # Format the inputs for ONNX Runtime
+    model_inputs = {
+        'input_ids':   [input_ids], 
+        'input_mask':  [input_mask],
+        'segment_ids': [segment_ids]
+        }
+                  
+    outputs = session.run(['start_logits', 'end_logits'], model_inputs)
+    logging.info("Post-processing")  
+
+    # Post process the output of the model into an answer (or an error if the question could not be answered)
+    results = postprocess(tokens, outputs[0], outputs[1])
+    logging.info(results)
+    return results
+
+
+if __name__ == '__main__':
+    init()
+
+    input = "{\"question\": \"What is Dolly Parton's middle name?\", \"context\": \"Dolly Rebecca Parton is an American singer-songwriter\"}"
+
+    run_pytorch(input)
+    print(run(input))
+```
+
+## Deploy model with ONNX Runtime through AzureML
+
+Now that we have the ONNX model and the code to run it with ONNX Runtime, we can deploy it using Azure ML.
+
```

Comment:
**Component diagram: AzureML deployment with ONNX Runtime (environment, ONNX model, and scoring code)**

Reply:
This is for screen readers so I think the punctuation might stall that